mhw_library/utils.py

```diff
@@ -1,15 +1,362 @@
+    threshold = detrended.groupby('time.month').quantile(threshold_val,dim=('time')) 
+    features_ssta = detrended.where(detrended.groupby('time.month')>=threshold, other=np.nan)
+    features_ssta = features_ssta[:,:,:].load()
+    
+    # masking
+    full_mask_land = features_ssta
+    full_masked = full_mask_land.where(full_mask_land != 0)
+    binary_out_afterlandmask=np.isfinite(full_masked)
+    newmask = np.isfinite(ds.SST[0,:,:,:][:])
+    
+    # blobs
+    Tracker = ocetrac.Tracker(binary_out_afterlandmask[:,:,:], 
+                              newmask, 
+                              radius=radius_val, 
+                              min_size_quartile=min_size_quartile_val, 
+                              timedim = 'time', 
+                              xdim = 'lon', 
+                              ydim='lat', 
+                              positive=True)
+    blobs = Tracker.track()
+    blobs.attrs
+    mo = Tracker._morphological_operations()
+    return detrended, blobs
+
+def combine_sst_and_labels(detrended: xr.DataArray, blobs: xr.DataArray) -> xr.Dataset:
+    """
+    Merge the given SST anomaly data array with the object identifier labels data array.
 
-def create_events_file(detrended, blobs):
-    '''
-    Merges SST anomalies with the MHW identifier labels.
     Parameters
     ----------
-    detrended : xarray.DataArray (time:1980, lat: 192, lon: 288)
-        Consists of SST anomalies.
-    blobs : xarray.DataArray 'labels' (time: 1980, lat: 192, lon: 288)
-        Consists of MHW identifier labels.
-    '''
+    detrended : xarray.DataArray of shape (time: 1980, lat: 192, lon: 288)
+        A data array containing SST anomalies.
+    blobs : xarray.DataArray of shape (time: 1980, lat: 192, lon: 288)
+        A data array containing object identifier labels.
+
+    Returns
+    -------
+    xarray.Dataset
+        A dataset containing the following data variables:
+        - 'SSTA': SST anomaly data array
+        - 'labels': object identifier labels data array
+    """
     detrended.name = 'SSTA'
-    return xr.merge([detrended, blobs])
\ No newline at end of file
+    return xr.merge([detrended, blobs])
+
+def count_objects(merged_xarray: xr.Dataset) -> int:
+    """
+    Counts the total number of objects in an event file.
+
+    Parameters
+    ----------
+    merged_xarray : xr.Dataset
+        A merged xarray containing the data variables 'SSTA' and 'labels'.
+
+    Returns
+    -------
+    int
+        The total number of objects in the 'labels' data variable, excluding the background.
+
+    Notes
+    -----
+    The 'labels' data variable should contain integer labels that identify connected regions in the 'SSTA' data variable.
+    The value 0 is reserved for the background and is not considered an object.
+    """
+    return len(np.unique(merged_xarray.labels)) - 1
+
+def calc_initialization(merged_xarray: xr.Dataset, mhw_id: int) -> Tuple[int, xr.DataArray, int]:
```

Comment:
calc_initialization -> cal_initial_state? I was a bit confused what was meant by initialization at first. or get_initial_state since it is not calculating much.